### DISCUSSION TITLE
Graceful shutdown

### DISCUSSION THREAD

**Discussion by lordigon** [arkmq-org/arkmq-org-broker-operator#814] [Q&A]
Hi all,
I would like to know if graceful shutdown is enabled by default into Artemis pods or if it must be enabled in some way via CRD.

Best regards

**Comment by gtully** [comment]
if a pod is gracefully stopped then the broker running in that pod will get a chance to shutdown on the first interrupt signal. So long as the grace period is sufficient, the broker will gracefully shutdown. there is nothing special in the CR to enable that behaviour. It is essentially the jvm responding to the kill signal that is part of pod termination of a running container
